src/VsExtension/NuGetPackage.cs

```diff
@@ -372,6 +368,52 @@ private void SetDefaultCredentialProvider()
                 // add the proxy to v2 proxy cache.
                 v2ProxyCache?.Add(proxy);
             };
+
+            NuGet.Protocol.Core.v3.HttpHandlerResourceV3.ProxyPassed = proxy =>
+            {
+                // add the proxy to v2 proxy cache.
+                v2ProxyCache?.Add(proxy);
+            };
+
+            var v2CredentialStoreType = typeof(NuGet.ICredentialCache).Assembly.GetType("NuGet.CredentialStore");
+            var property = v2CredentialStoreType?.GetProperty(
+                "Instance",
+                BindingFlags.Static | BindingFlags.Public);
+            var v2CredentialStore = property?.GetValue(obj: null) as NuGet.ICredentialCache;
+
+            NuGet.Protocol.Core.v3.HttpHandlerResourceV3.PromptForCredentials = uri =>
```

Comment:
Looks good to me. I especially like that we have the separate PromptForCredentials from PromptForProxyCredentials. Plus v2 CredentialStore and v3 credentails are fully synced.